File: sprint1/By lessons/lesson7code6/lesson7code6.py
def get_least_primes_linear(n):
    lp = [0] * (n + 1)
    primes = []
    for i in range(2, n + 1):
        if lp[i] == 0:
            lp[i] = i
            primes.append(i)
        for p in primes:
            x = p * i
            if (p > lp[i]) or (x > n):
                break
            lp[x] = p
    return primes, lp

File: sprint1/By languages/swift/lesson7code6.swift
func getLeastPrimesLinear(_ n: Int) -> ([Int], [Int]) {
    var lp = [Int](repeating: 0, count: n + 1)
    var primes = [Int]()
    for i in 2...n {
        if lp[i] == 0 {
            lp[i] = i
            primes.append(i)
        }   
        for p in primes {
            let x = p * i
            if p > lp[i] || x > n {
                break
            }
            lp[x] = p
        }
    }
    return (primes, lp)
}